### DISCUSSION TITLE
Compiling on Raspberry Pi 64 bit

### DISCUSSION THREAD

**Discussion by 00RedD00** [emporia-vue-local/esphome#147] [Show and tell]
I just flashed by emporia, not installed yet, and like others I wasted a lot of time because the pi would not compile.
After of lot of trials, I got it to compile simply by changing "platform_version: recommended" to "platform_version: 5.3.0".
Can someone that has the vue installed try to compile with a pi and check if everything works? 
I so adding this at the end of the doc under the "I'm using a 64-bit Pi & can't compile!" section would save others a lot of  time.

**Comment by flaviut** [comment]
I've added a link to this discussion. Thanks for the feedback!

**Reply by 00RedD00** [reply]
I realise I was not super accurate in my description, so in case, this is the actual config that compiles for me: 
esp32:
  board: esp32dev
  framework:
    type: esp-idf
    platform_version: 5.3.0
    sdkconfig_options:
      CONFIG_COMPILER_OPTIMIZATION_SIZE: y
      
    No idea if the 2 last lines do anything

**Comment by excalbian** [comment]
Just wanted to report that I was able to compile successfully on a Pi-4 64-bit (within the esphome add-on of my HomeAssistant instance) without changing the platform version. Perhaps platformio has added support for 64-bit ARM (Raspberry Pi) in a more recent release?

```
Linking /data/vue-alpha/.pioenvs/vue-alpha/firmware.elf
RAM:   [=         ]   9.3% (used 30440 bytes from 327680 bytes)
Flash: [=====     ]  46.3% (used 848833 bytes from 1835008 bytes)
Building /data/vue-alpha/.pioenvs/vue-alpha/firmware.bin
Creating esp32 image...
Successfully created esp32 image.
esp32_create_combined_bin(["/data/vue-alpha/.pioenvs/vue-alpha/firmware.bin"], ["/data/vue-alpha/.pioenvs/vue-alpha/firmware.elf"])
Wrote 0xe0a30 bytes to file /data/vue-alpha/.pioenvs/vue-alpha/firmware-factory.bin, ready to flash to offset 0x0
======================== [SUCCESS] Took 900.06 seconds ========================
INFO Successfully compiled program.
```

There was one warning, but doesn't seem to have caused any issues (I'm not sure, may also show up on other environments):
```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
esphome 2023.6.2 requires pyparsing>=3.0, but you have pyparsing 2.3.1 which is incompatible.
```

My versions:

- Home Assistant 2023.6.3
- Supervisor 2023.06.2
- Operating System 10.3
- Raspberry Pi 4 / rpi 4-64
- Esphome 2023.6.2

I'm using a [templated version of the config](https://github.com/emporia-vue-local/esphome/discussions/179) from the readme, here are the `esphome` and `esp32` sections:

```
esphome:
  name: ${devicename}
  friendly_name: ${devicename_upper}

esp32:
  board: esp32dev
  framework:
    type: esp-idf
    version: recommended
```
